### DISCUSSION TITLE
Unable to access next_auth schema while use SupabaseAdapter with NextAuth.js

### DISCUSSION THREAD

**Discussion by dhairyathedev** [supabase/supabase#20021] [Questions]
Error, I am getting:
`Object { code: "42501", details: null, hint: null, message: "permission denied for schema next_auth" }`

My code:
```js
export async function getUserDetails(id, sessionToken) {
  const { data, error } = await supabase(sessionToken).schema("next_auth").from("accounts").select("*").eq("id", id).single();
  if (error) {
    console.log(error);
  }
  return data;
}
```
I want to access the next_auth schema, to access the user's refresh token from my backend server from the database.

**Comment by GaryAustin1** [comment]
As mentioned in Discord, this is likely you have not set grants for authenticated on the schema and tables.  The  guide from auth.js you showed there set the grants only for service_role and postgres.

Normally you would not want your auth table accessed by users though.  A better solution might be to leave the grants and use a security definer Postgres function called by RPC to get the data in a secure fashion.
